tests/integration/moderation.test.ts

```diff
@@ -1,184 +1,187 @@
-    const result = await reportRef.get()
-    expect(result.exists).toBeTruthy()
-    expect((result.data() as Report).reason).toBeDefined()
+    const rep1 = await getThisReport()
+    expect(rep1).toBeDefined()
+    expect(rep1!.id).toEqual(reportId)
+    // clean up
+    await removeThisTestimony()
+    await removeThisReport()
   })
 })
 
 describe("action functions", () => {
   it("set file report resolution via resolveReport", async () => {
-    // await expect(signInTestAdmin()).resolves.toBeDefined()
-
-    const result = await reportRef.get()
-    expect(result.exists).toBeTruthy()
+    // set up
+    const { tid, getThisTestimony, whereIsThisTestimony, removeThisTestimony } =
+      await createNewTestimony(uid, "H1002")
+    const { reportId, getThisReport, removeThisReport } = await createNewReport(
+      uid,
+      tid
+    )
 
-    expect((result.data() as Report).reason).toBeDefined()
-    expect((result.data() as Report).reportId).toBeDefined()
+    const report = await getThisReport()
+    expect(report).toBeDefined()
+    expect(report?.reason).toBeDefined()
+    expect(report?.reportId).toBeDefined()
 
-    const reportId = result.data()?.id
     const resolution = "remove-testimony" /// need to test both allow and remove
     const reason = "important reason"
 
-    // expect(
-    //   (await resolveReport({
-    //     reportId,
-    //     resolution,
-    //     reason
-    //   })).data.status
-    // ).toEqual("success" )
+    const result = await resolveReport({
+      reportId,
+      resolution,
+      reason
+    })
+
+    expect(result.data.status).toEqual("success")
+
+    //clean up
+    await removeThisTestimony()
+    await removeThisReport()
   })
 
-  it.skip("move testimony from published to archived via deleteTestimony https", async () => {
+  it("moves testimony from published to archived via deleteTestimony endpoint", async () => {
     // set up
-    // const { user, testimony, report } = createFakeTestimonyReport()
-    // const localReportRef = testDb.doc(`reports/${report.id}`)
-    // const localTestimonyRef = testDb.doc(
-    //   `users/${user.uid}/publishedTestimony/${testimony.id}`
-    // )
-    // await localTestimonyRef.set(testimony)
-    // await localReportRef.set(report)
+    const { tid, getThisTestimony, whereIsThisTestimony, removeThisTestimony } =
+      await createNewTestimony(uid, "H1002")
+    const { reportId, getThisReport, removeThisReport } = await createNewReport(
+      uid,
+      tid
+    )
     const refresh = jest.fn()
-    await expect(signInTestAdmin()).resolves.toBeDefined()
+    await signInTestAdmin()
 
-    const admin = testAuth.getUser(adminUid)
-    expect(admin).toBeDefined()
+    const [report, testimony] = await Promise.allSettled([
+      getThisReport(),
+      getThisTestimony()
+    ])
 
-    expect((await testimonyRef.get()).exists).toBeTruthy()
-    expect((await reportRef.get()).exists).toBeTruthy()
+    report.status === "fulfilled"
```

Comment:
Will this skip `expect` when it's rejected? If we don't have any `expect` calls I think the test might actually pass, which I don't think we want.

Reply:
Yes, reject will skip expect there, but will get caught by the expect at the end, which is all this one cares about. In between is just redirecting & logging errors that keep the test from running through. Maybe this test isn't a keeper, but it was very helpful for what I was doing at the time 🤷